src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,67 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> INTERFACE_IMPL = Map.of(
+            FileReaderService.class, FileReaderServiceImpl.class,
+            ProductService.class, ProductServiceImpl.class,
+            ProductParser.class, ProductParserImpl.class);
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public Object getInstance(Class<?> interfaceClass) {
+        Class<?> implClass = findImplementation(interfaceClass);
+        checkComponentAnnotation(implClass);
+        Field[] declaredFields = implClass.getDeclaredFields();
+        Object classImplInstance = createNewInstance(implClass);
```

Comment:
![image](https://github.com/mate-academy/jv-dependency-injection/assets/82539553/a6930369-c1e4-443c-a038-85d403a10d7c)
Create map where you can store created instances to not create the same class several times.